pkg/types/validation/installconfig.go

```diff
@@ -155,6 +155,35 @@ func ValidateInstallConfig(c *types.InstallConfig, usingAgentMethod bool) field.
 		}
 	}
 
+	if c.Publish == types.MixedPublishingStrategy {
+		switch platformName := c.Platform.Name(); platformName {
+		case azure.Name:
+		default:
```

Comment:
Nit: Maybe `if` looks better here?
```suggestion
		if platformName := c.Platform.Name(); platformName != azure.Name {
```